registry/client.go

```diff
@@ -62,6 +65,31 @@ type credentials struct {
 	password string
 }
 
+type schemaInfoPayload struct {
+	Schema  string `json:"schema"`
+	ID      int    `json:"id"`
+	Version int    `json:"version"`
+	Subject string `json:"subject"`
+}
+
+func (s *schemaInfoPayload) Parse() (SchemaInfo, error) {
+	info := SchemaInfo{
+		ID: s.ID,
+		Version: s.Version,
+		Subject: s.Subject,
+	}
+	schema, err := avro.Parse(s.Schema)
+	info.Schema = schema
+	return info, err
+}
```

Comment:
```suggestion
func (s *schemaInfoPayload) Parse() (info SchemaInfo, err error) {
	info = SchemaInfo{
		ID: s.ID,
		Version: s.Version,
		Subject: s.Subject,
	}
	info.Schema, err = avro.Parse(s.Schema)
	return info, err
}
```

Reply:
I actually had something like this originally, but I get a compile error saying that `err` is not defined. The following works though:
```go
func (s *schemaInfoPayload) Parse() (SchemaInfo, error) {
	var err error
	info := SchemaInfo{
		ID: s.ID,
		Version: s.Version,
	}
	info.Schema, err = avro.Parse(s.Schema)
	return info, err
}
```

Would you prefer this?

Reply:
oh whoops, did not see you named the return types. That fixes it. Thanks!